### DISCUSSION TITLE
xpack support in cdt-cloud

### DISCUSSION THREAD

**Discussion by vinodappu** [eclipse-cdt-cloud/cdt-cloud#30] [General]
I’m interested in https://marketplace.visualstudio.com/items?itemName=ilg-vscode.xpack , which I think should be available in open-vsx.

Is it possible to use VS-Code extensions directly installing from VS-code marketplace in Theia (for some learning purpose)? I’ve tried to copy extension files to /plugins folder, which worked and gave UI options but didn’t work as in VS-Code. I’m thinking it’s because of dependencies. Can I use VSX_REGISTRY_URL and install extensions directly from market place using https://marketplace.visualstudio.com/vscode instead http://open-vsx.org/ ?

**Comment by planger** [comment]
As far as I know, it is legally not allowed to directly connect from a Theia-based application to the VS Code marketplace. You should be able though to download extensions from there as VSX file and manually install them in your Theia-based application. Or, if they have a compliant open-source license, [publish them in OpenVSX](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions) and download them from there.

I guess, the reason why `ilg-vscode.xpack` doesn't work as expected is because it [depends on](https://github.com/xpack/vscode-xpack-extension-ts/blob/fef3163c2ede8a42a127544ec8cddd9057348f0a/package.json#L42) `ms-vscode.cpptools`, which in turn is closed source and not available on OpenVSX. I don't know whether `ms-vscode.cpptools` works in Theia if you'd manually install it, for the sake of testing.

Another aspect to consider is that `ilg-vscode.xpack` [requires VS Code API 1.71.0](https://github.com/xpack/vscode-xpack-extension-ts/blob/master/package.json#L45). Theia currently supports up to 1.68.2 on master (soon [1.70.2](https://github.com/eclipse-theia/theia/issues/12028)). So depending on which kind of the missing API `ilg-vscode.xpack` actually uses, it may still work.

On a first glance, it looks like the functionality `ilg-vscode.xpack` provides is somewhat going in a similar direction as [clangd-contexts](https://github.com/eclipse-cdt-cloud/clangd-contexts) and/or [CMake presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html). So it'd be interesting to learn more about your requirements to see if those would match your needs, and if not, whether CDT Cloud should provide a similar functionality, without a dependency on a closed-source extension, such as `ms-vscode.cpptools`.

Can you maybe elaborate on your requirements to see if it matches and whether others would have similar requirements?

Thanks!
